soc/riscv/riscv-privilege/nuclei/gd32vf103/soc.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2021 Tokita, Hiroshi <tokita.hiroshi@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file SoC configuration macros for the GigaDevice GD32V processor
+ */
+
+#ifndef __RISCV_GIGADEVICE_GD32V_SOC_H_
+#define __RISCV_GIGADEVICE_GD32V_SOC_H_
+
+#include <soc_common.h>
+#include <devicetree.h>
+
+/* Timer configuration */
+#define RISCV_MTIME_BASE             (0xD1000000UL + 0)
+#define RISCV_MTIMECMP_BASE          (0xD1000000UL + 0x8)
+
+/* lib-c hooks required RAM defined variables */
+#define RISCV_RAM_BASE               DT_SRAM_BASE_ADDRESS
+#define RISCV_RAM_SIZE               KB(DT_SRAM_SIZE)
+
+#ifndef _ASMLANGUAGE
+#include <toolchain.h>
+#include <gd32vf103.h>
+
+#endif  /* !_ASMLANGUAGE */
+
+#define USART_STAT0 USART_STAT
```

Comment:
You should add this at beginning of usart driver.

Reply:
ref https://github.com/zephyrproject-rtos/zephyr/pull/40283/files#diff-b489f17675772dbd2ed976303b084aa13f70f9926606ca202eb3755e9db19fccR45

Reply:
Use usart_gd32_stat from #40283.